### ISSUE TITLE
Not able to replicate results on dev branch

### ISSUE THREAD

**Issue by bglearning** [allenai/bi-att-flow#68]
We are trying to replicate the results using the dev branch but not able to do so. 

![results](https://user-images.githubusercontent.com/4636315/34595436-11710fba-f200-11e7-8bd2-f35d4d1de922.png)

The scores plateau quickly around 43 for F1 and 37 for EM after the ~2000th step. 

We ran it on [p2.xlarge on AWS](https://aws.amazon.com/blogs/aws/new-p2-instance-type-for-amazon-ec2-up-to-16-gpus/) which has 12GiB of GPU memory. 

This happens for both the [1.2 compatible version](https://github.com/allenai/bi-att-flow/commit/a9dea34548a2e230b92d8d8de5018f58853f23c7) and the [1.1 one](https://github.com/allenai/bi-att-flow/commit/14ee0e46ce23ab031fda1ba7f2aa3975d98486ee).

**Comment by liuyaqiu** [comment]
You should be careful to the parameter "init_lr",  which is set in basic.cli module. In the TF r0.11 version, this parameter is set to 0.5, but in the TF r1.1 version, it is set default to 0.001. I think this is contributing to the problem, and you hadn't pay attention to this.

**Comment by bglearning** [comment]
@liuyaqiu Do you mean we should tinker with it? We are running the dev branch as is. Shouldn't that work?

**Comment by liuyaqiu** [comment]
@bglearning You should set the "init_lr" to 0.5, I think this will solve the problem. If the learning rate is too small, the model may fall into a local optimal and this will contribute to overfit.

**Comment by bglearning** [comment]
Tried setting the learning rate to 0.1. It performed badly from the get-go. It's probably not that.

**Comment by Leeyouxie** [comment]
@liuyaqiu Did you run this model on master branch, and get 67.7% for em and 77.3% for F1?

**Comment by liuyaqiu** [comment]
@Leeyouxie No, I recently run three times using Adadelta optimizer with init_lr 0.5, then get 65.5 EM and 75.5 F1. I think I can't get the referenced result in the paper.

**Comment by Leeyouxie** [comment]
@liuyaqiu , Thank for your reply. I ran this model without modifying codes but 30 for batch size. However,  I just get about 0.64 for EM and about 0.75 for F1.

**Comment by dengyuning** [comment]
Did you solve this problem? Cause I meet the same problem as you.

**Comment by bglearning** [comment]
@dengyuning Ya. Sort of. Kind of embarrassing but turns out it wasn't a problem per se. Just that the evaluation done during training didn't generate answers for all questions. (2570 questions remained unanswered.) We didn't dig into it so don't know exactly why it is so. 

Anyway, therefore, if you run the official evaluation script like so:

```
python squad/evaluate-v1.1.py ~/data/squad/dev-v1.1.json out/basic/00/answer/dev-001000.json
```
You should see a bunch of `Unanswered question ... will receive score 0.` 

Instead if you run the full test
```
python -m basic.cli
```
You should see the desired performance.
